roombasim/pittras/task/takeoff_task.py

```diff
@@ -0,0 +1,47 @@
+
+import numpy as np
+
+import roombasim.config as cfg
+
+from roombasim.ai import Task
+
+class TakeoffTaskState:
+    init = 0
+    request_arm = 1
+    pause_before_takeoff = 2
+    takeoff = 3
+    ascend = 4
+    ascend_angle = 5
+    done = 6
+    failed = 7
+
+class TakeoffTask(Task):
+
+    def __init__(self):
+        # null velocities
+        self.zero_xy_vel = np.array([0,0], dtype=np.float64)
+
+        # state machine
+        self._state = TakeoffTaskState.init
+
+    def update(self, delta, elapsed, state_controller, environment):
+        drone_state = state_controller.query('DroneState', environment)
+
+        height = drone_state['z_pos']
+        control_z_vel = drone_state['z_vel']
+
+        # Pause before ramping up the motors
+        if elapsed > cfg.PITTRAS_DELAY_BEFORE_TAKEOFF * 1000:
+            # Check if we reached the target height
+            if height < cfg.PITTRAS_TAKEOFF_COMPLETE_HEIGHT:
+                # Ascend if below the target height
+                self._state = TakeoffTaskState.ascend
+
+                control_z_vel = cfg.PITTRAS_TAKEOFF_VELOCITY
+            else:
+                # Stop ascending if above the target height
+                self._state = TakeoffTaskState.done
+
+                control_z_vel = 0
+
+        environment.agent.control(self.zero_xy_vel, 0, control_z_vel)
```

Comment:
You can pass a list for the xy_acceleration rather than np.array since it gets upconverted anyways on the drone side. E.g. `[0,0]` instead of `self.zero_xy_vel`.

Reply:
Additionally, I changed the signature of `Drone.control` in the part you rebased to accept: xy acceleration vector, yaw *acceleration* vector and z *acceleration* vector. Although looking at the code in iarc_motion, it seems like using the old yaw *velocity* and z *velocity* would work fine.

Reply:
Ok, I reverted the control method back to the original so the above call should work as expected

Reply:
Oh thanks for that, I didn't notice that you changed it to take an acceleration value. I worked on before that change was committed.